pkg/clients/aws.go

```diff
@@ -517,6 +517,21 @@ func LateInitializeInt64Ptr(in *int64, from *int64) *int64 {
 	return from
 }
 
+// Float64Value returns the value of the float64 pointer passed in or
+// 0 if the pointer is nil.
+func Float64Value(v *float64) float64 {
+	return aws.Float64Value(v)
+}
```

Comment:
It looks like we already do this for some other types, but what is the rationale for wrapping the AWS SDK function here? Also, will we actually need `resource.Quantity` to `float64` given that CRDs do not support `float64`?

Reply:
> what is the rationale for wrapping the AWS SDK function here?

I am not sure TBH, just copied the content of `Int64Value` to get EFS generated code to work for TBS today.

> Also, will we actually need resource.Quantity to float64 given that CRDs do not support float64?

It seems to give `type: number` in OpenAPIv3Schema but I haven't tried giving that field a value. Let's keep this unmerged for now and I'll see if `float64` is OK or a change is needed in ACK to use `resource.Quantity` strictly

Reply:
Seems it was accepting integers in practice anyway, so I ignored that field and used `*int64` but we still need some helpers to do the casting from and to `*float64`. Updated the PR to remove the `LateInitializeFloat64Ptr` function, which is not needed now.

Reply:
@muvaf I think I am still confused on why we would use `Float64Value` here instead of just `aws.Float64Value` directly?

Reply:
Personally, I think we should have the actual implementations here in this file without depending on AWS but I was in a rush when I opened this PR. Let me change all these small functions to have the actual implementations.

Reply:
Haha just saw my own note `// TODO(muvaf): is this really meaningful? why not implement it?`

Reply:
>why we would use Float64Value here instead of just aws.Float64Value directly?

Sorry I realized I didn't answer this. There are cases where we need to give `0` if the value is `nil` since they are required, hence we needed the `FieldOption` struct in for example `String` function in this file. But that's not implemented in some of the resources, so I'm updating that now.